Lib/ldap/cidict.py

```diff
@@ -50,6 +50,9 @@ def get(self,key,failobj=None):
     except KeyError:
       return failobj
 
+  def __iter__(self):
```

Comment:
``UserDict.IterableUserDict`` and ``collections.Mapping`` always provide ``__iter__``

Reply:
True, but it is a different implementation. 

https://github.com/python/cpython/blob/2.7/Lib/UserDict.py#L113-L114

The Python one iterates over the keys of stored data (`self.data.keys()`). For cidict, this will be the lowercased versions. But for cidict, we should use the original vales.

I could drop this `__iter__` if the library were changed to:

```py
  def __setitem__(self,key,value):
    lower_key = key.lower()
    self._keys[key] = lower_key
    self.data[key] = value

  def __getitem__(self,key):
    key = self._keys.get(key.lower(), key)
    return self.data[key.lower()]
```

Otherwise, this method will need to stay. What do you think?